lib/beaker/host.rb

```diff
@@ -571,7 +571,9 @@ def do_rsync_to from_path, to_path, opts = {}
       @logger.notify "rsync: localhost:#{from_path} to #{hostname_with_user}:#{to_path} {:ignore => #{opts[:ignore]}}"
       result = Rsync.run(from_path, to_path, rsync_args)
       @logger.debug("rsync returned #{result.inspect}")
-      result
+
+      return result if result.success?
+      raise Beaker::Host::CommandFailure, result.error
```

Comment:
@Dakta please add a `@return` & `@raise` yard statement to the method, so we'll be better informed as to the different return types depending on the success of rsync execution.

To what you mentioned about nested exceptions (although it sounds like `Rsync::Result` will fail without throwing an exception), some quick googling pointed me to [this article](http://blog.honeybadger.io/nested-errors-in-ruby-with-exception-cause/), which makes it sound like exceptions will automatically get chained together and can be inspected using the `exception.cause` method.

Reply:
Done.